src/main/java/org/apache/commons/imaging/formats/jpeg/decoder/JpegDecoder.java

```diff
@@ -376,16 +374,14 @@ private void readMCU(final JpegInputStream is, final int[] preds, final Block[]
                                 is,
                                 huffmanACTables[scanComponent.acCodingTableSelector]);
                         final int ssss = rs & 0xf;
-                        final int rrrr = rs >> 4;
-                        final int r = rrrr;
```

Comment:
It does look odd, but if you look at the comments near this code, they are pointing to which page of a document the author was referencing when writing it. These variable names, though obscure or redundant, are more helpful for devs familiar with that document.

See this Python examples: 

- https://github.com/Exaphis/python-jpeg-decoder/blob/849c3d0bb32355e44a40364ea0498e6f59443716/jpeg.py#L302
- https://github.com/yohhoy/picojdec/blob/316ef275c757bb67d4398980660f42915f67901c/picojdec.py#L152

Or these excerpts from a specification document:

![image](https://user-images.githubusercontent.com/304786/124344537-f7408200-dc26-11eb-98a2-6f27f03d4827.png)

![image](https://user-images.githubusercontent.com/304786/124344560-14755080-dc27-11eb-81e6-29fbe3a6ebb3.png)

I use Eclipse + IntelliJ (and other tools like CPD, SpotBugs, stan4j, sonarqube, etc) but for [imaging] I ignore several issues to avoid reducing the clarity of the code for future developers familiar with image processing :+1: 

So sorry for turning down changes like this @arturobernalg , even though you are correct some variables are redundant.